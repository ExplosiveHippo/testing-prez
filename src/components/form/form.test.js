import React from "react";
import { shallow } from 'enzyme';
import Form from './form';

describe("<Form />", () => {
  const inventory = { name: "Fresh Prince", available: 6 };
  const form = shallow(<Form inventory={inventory} />);
  
  it("should render", () => {
    expect(form).toMatchSnapshot();
  });

  it('should be initialized with empty value ', () => {
     expect(form.state().value).toEqual("");
  });

  it("should update form state on input change", () => {
    form.find("#input").simulate("change", { target: { value: "5" } });
    expect(form.state().value).toEqual("5");
  });

  it("should disable button if input is empty", () => {
    form.find("#input").simulate("change", { target: { value: "" } });
    expect(form.find("#button").prop("disabled")).toEqual(true);
  });
  

  it("should limit quantity to available quantity", () => {
    const inventory = { name: "Fresh Prince", available: 6 };
    form.find("#input").simulate("change", { target: { value: "50" }});
    expect(form.find("#button").prop("disabled")).toEqual(true);

  });

  
});