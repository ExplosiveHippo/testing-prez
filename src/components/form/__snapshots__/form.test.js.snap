// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    inventory={
        Object {
            "available": 6,
            "name": "Fresh Prince",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          id="input"
          onChange={[Function]}
          val=""
/>,
        <button
          disabled={false}
          id="button"
>
          Add To Cart
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "input",
          "onChange": [Function],
          "val": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add To Cart",
          "disabled": false,
          "id": "button",
        },
        "ref": null,
        "rendered": "Add To Cart",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="input"
            onChange={[Function]}
            val=""
/>,
          <button
            disabled={false}
            id="button"
>
            Add To Cart
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "input",
            "onChange": [Function],
            "val": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add To Cart",
            "disabled": false,
            "id": "button",
          },
          "ref": null,
          "rendered": "Add To Cart",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
